spinalcordtoolbox/deepseg/models.py

```diff
@@ -356,3 +379,19 @@ def get_metadata(folder_model):
     with open(fname_metadata, "r") as fhandle:
         metadata = json.load(fhandle)
     return metadata
+
+
+def find_ensemble_subfolders(path_model):
+    """
+    Search for the presence of model subfolders within the main model folder. If they exist,
+    then the model folder is actually an ensemble of models, so return a list of folders.
+
+    :param path_model: Absolute path to folder that encloses the model files.
+    :return: list or str: Either a list of ensemble subfolders, or the original model folder path.
```

Comment:
Just to sketch out a idea of what this might look like:

- Because seed names are problematic, we could perhaps ditch the names, and instead use a "list of `<mirror URL lists>`" instead of a "dict of `<mirror URL lists>`" in the `MODELS` definition.
- For downloading the model ensemble, we would just check for the presence of sublists (instead of checking for dicts).
- Then, as you suggest, `path_model` would get parsed as `path_models` (i.e. a list of all valid model folders)
- Then, we would _always_ use the "inference then average" function, no matter if there are 1 or 2+ models (since computing `np.mean` with a single image is just that image)
- Meaning that we have a single code path regardless of the number of model paths?

Reply:
I think this is a fine suggestion? I'd be curious to try implementing it to see how it compares. :)

Reply:
> But the existence or non-existence of seed names is a significant difference, unfortunately.

As you point out in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4042#discussion_r1113337222, seed names are only relevant for logging (to help distinguish the different inference logs for different seeds).

But, looking at the lines that you highlighted in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4042#discussion_r1113337222:

```python
       name_seed = Path(path_model).parts[-2]
        logger.info(f"\nRunning inference using model '{name_model}' ({name_seed})...")
```

Seed names are _inferred from the model path_. So, we could always just... only output these logs if `len(model_path) > 1`? That way, we only parse the path for seed names if we _know_ we're using an ensemble.

Reply:
Hmm. I tried implementing this suggestion in a49c18e978f6ff33ec28ca5da89a2ad5534b6914.

This does indeed simplify the code path outside the `segment_volume` function calls (since we use the same function call regardless of whether we have 1 or for 2+ models). However, in doing so, we make the `segment_volume` function itself a little more complex, since we have to account for both ensemble and non-ensemble cases. 

I'm happy to keep the changes, but I'm curious what your thoughts are on the changes here.

Reply:
Thanks for trying out my suggestion. You're right that it ends up being roughly equivalent in terms of complexity. In the end, I have no real preference between the two versions, so feel free to pick the one you prefer.

(One slight behavior difference between the two versions, in case that helps you decide: if somehow we had an ensemble model with a single named seed, this new version would skip printing out the seed name, and the averaging step.)

Reply:
> (One slight behavior difference between the two versions, in case that helps you decide: if somehow we had an ensemble model with a single named seed, this new version would skip printing out the seed name, and the averaging step.)

I think I like this new behavior, especially given the above.